code/ATMOSPHERICS/hvac/geyser.dm

```diff
@@ -0,0 +1,89 @@
+#define GEYSER_TEMP 40
+#define GEYSER_POWER 40000
+#define GEYSER_HEAT_RANGE 3
+#define GEYSER_WARM_RATE 25
+
+/obj/structure/geyser
+	name = "thermal geyser"
+	desc = "A geyser is a type of spring that is caused by the upward force of heat attemping to escape outward from a mineral body."
+	density = 1
+	anchored = 1
+	icon = 'icons/misc/beach.dmi'
+	icon_state = "geyser"
+
+	var/stability = "usual"
+
+	var/heat_range = 3
+	var/warm_rate = 25
+	var/smoke_probability = 40
+	var/preparing_smoke = 0
+
+/obj/structure/geyser/process()
+	if (preparing_smoke)
+		return // We're about to exhaust smoke
+	if(prob(smoke_probability))
+		puff_smoke()
+	var/turf/T = loc
+	if(!istype(T))
+		return //Not on a turf
```

Comment:
Is there any reason you check for a null turf _after_ the _puff_smoke()_ call?

Also you can simply do `if(!isturf(loc))`

Reply:
Gerysers are `obj` and can, probably, be put inside other atoms in specific situations.

Reply:
But would that make any sense? I just don't see it